examples/src/main/scala/mhtml/examples/Chosen.scala

```diff
@@ -29,124 +26,117 @@ object Searcheable {
+
+    val rxSelected: Rx[Option[T]] =
+      rxHighlightedCandidate.sampleOn(enterPressedEvents) |+| clickSelectionEvents
+
+    val rxChosenOptions: Rx[Node] =
+      (rxIndex |@| rxFocused |@| queryEvents |@| rxFilteredCandidates).map {
+        case(index, focus, query, fcand) =>
+          def bounds(i: Int): Int = if (fcand.size > maxCandidates) i max 0 else 0
+          val toDrop = bounds(index - 3)
+          val remain = bounds(fcand.size - (toDrop + maxCandidates))
+          val style  = if (focus)  None else Some("display: none")
+          val itemsBefore = if (toDrop == 0) None else Some(<li>{ toDrop } more items...</li>)
+          val itemsAfter  = if (remain == 0) None else Some(<li>{ remain } more items...</li>)
+          val listItems   =
+            fcand.zipWithIndex.slice(toDrop, toDrop + maxCandidates).map { case (candidate, i) =>
+              <li class={ if (i == index) "chosen-highlight" else "" }>
+                <a onclick={ () => clickSelectionEvents := Some(candidate) }>
+                  { underline(Searcheable[T].show(candidate), query.toLowerCase) }
+                </a>
+              </li>
+            }
+          <div class="chosen-options">
+            <ul style={ style }>
+              { itemsBefore }
+              { listItems }
+              { itemsAfter }
             </ul>
           </div>
-      div -> candidates
-    }
-    val rxCandidates: Rx[Seq[T]] = rxCandidatesWithApp.map(_._2)
-    val highlightedCandidate: Rx[T] = {
-      val filtered = Var[T](rxCandidates.value.head)
+      }
 
-      (for { index <- rxIndex; candidates <- rxCandidates } yield {
-        candidates.zipWithIndex.find(_._2 == index).map(_._1)
-      }).foreach(_.foreach(filtered.:=))
+    var cancelableSelectionHandler = Cancelable.empty
+    def selectionHandler(node: HTMLInputElement): Unit =
+      cancelableSelectionHandler =
+        rxSelected.impure.foreach(c => node.value = c.map(Searcheable[T].show).getOrElse(""))
 
-      filtered
-    }
-    // event handlers
-    val onkeyup = { e: KeyboardEvent =>
+    def onkeydown(e: KeyboardEvent): Unit =
       e.keyCode match {
-        case KeyCode.Up =>
-          rxIndex.update(x => Math.max(x - 1, 0))
-          rxFocused := true
-        case KeyCode.Down =>
-          rxCandidates.foreach { candidates =>
-            rxIndex.update(x => Math.min(x + 1, candidates.length - 1))
-          }.cancel()
-          rxFocused := true
-        case KeyCode.Enter =>
-          highlightedCandidate.foreach(setCandidate).cancel()
-        case _ =>
+        case KeyCode.Up    => arrowPressedEvents := -1
+        case KeyCode.Down  => arrowPressedEvents := +1
+        case KeyCode.Enter => enterPressedEvents := (())
+        case _ => ()
       }
-      ()
-    }
-    val onblur = { _: Event =>
-      js.timers.setTimeout(300)(rxFocused := false)
-      ()
-    }
+
+    // @olfa: This implementation does not really makes sense as it's a tick
```

Comment:
As written, this even triggered 400ms, 800ms, 1200ms ... after the page loaded. Which means, if the text field lost focus at millisecond 700, is would get "unfocused" only 100ms after that.